ha/k8s_setup/ha_start.py

```diff
@@ -26,44 +29,65 @@ def usage(prog: str):
+            f"usage: {prog} [-h] <-s/--services service to start> <-c/--config url> [-d/--debug [0|1]]...\n"
             f"where:\n"
-            f"--services health_monitor, fault_tolerance, k8s_monitor\n"
-            f"--config   Config URL\n")
+            f"-s --services health_monitor, fault_tolerance, k8s_monitor\n"
+            f"[-c --config   Config file URL : yaml://<file_path>] E.g. yaml:///etc/cortx/cluster.conf\n"
+            f"[-d --debug   [level]] (if level is not provided, then it considered as 1)"
+            f"\n\t\t0 : No debug.\n\t\t1 : Auto restarts child process if it dia.", file=sys.stderr, flush=True)
 
 def handle_signal(signum, frame):
     """
     Signal handler call beck function, to forward the received signal to driver process
     """
-    sys.stdout.write(f"Signal {signum} at frame {frame} received.")
-    sys.stdout.write(f"Sending {signum} to driver process pid: {driver_process.pid}.")
+    print(f"Signal {signum} at frame {frame} received.", flush=True)
+    print(f"Sending {signum} to driver process pid: {driver_process.pid}.", flush=True)
     driver_process.send_signal(signum)
 
-try:
-    if args.services not in service_entry_mapping:
-        sys.stdout.write('Please provide a valid service name')
-        usage('ha_start')
-        sys.exit(22)
-    # setting Signal handlers for all standered signals
-    signal.signal(signal.SIGABRT, handle_signal)
-    signal.signal(signal.SIGFPE, handle_signal)
-    signal.signal(signal.SIGILL, handle_signal)
-    signal.signal(signal.SIGINT, handle_signal)
-    signal.signal(signal.SIGSEGV, handle_signal)
-    signal.signal(signal.SIGTERM, handle_signal)
-
+def start_driver_process():
+    global driver_process
     driver_process = Popen(['/usr/bin/python3', service_entry_mapping[args.services]], shell=False, stdout=PIPE, stderr=STDOUT) # nosec
+    print(f"The driver process with pid {driver_process.pid}, and args {driver_process.args} started successfully.", flush=True)
+
+def main(argv: list):
+    try:
+        print(str(args))
+        if args.debug not in [0,1]:
+            sys.stderr.write('Please provide a valid debug value.')
+            usage('ha_start')
+            sys.exit(22)
+        if args.services not in service_entry_mapping:
+            print('Please provide a valid service name', flush=True)
+            usage('ha_start')
+            sys.exit(22)
+        # setting Signal handlers for all standered signals
+        signal.signal(signal.SIGABRT, handle_signal)
+        signal.signal(signal.SIGFPE, handle_signal)
+        signal.signal(signal.SIGILL, handle_signal)
+        signal.signal(signal.SIGINT, handle_signal)
+        signal.signal(signal.SIGSEGV, handle_signal)
+        signal.signal(signal.SIGTERM, handle_signal)
+
+        start_driver_process()
+
+        # Poll process.stdout to show stdout live
+        while True:
+            output = driver_process.stdout.readline()
+            if driver_process.poll() is not None:
```

Comment:
in below removed code this if condition breaking while loop but here you are continuing loop and break in else part, can u please check once again.

Reply:
ys because in case the auto-restart option is set by --debug flag will let continue to restart the process no matter why the child process was exited, killed, exit with an unhandled exception/crash.
so this is the aim so no need to break in this case.

Reply:
got it thanks.